arches_component_lab/views/api/relatable_resources.py

```diff
@@ -34,44 +34,35 @@ def get(self, request, graph, node_alias):
             "name", "graphid"
         )
 
-        resources = ResourceInstance.objects.filter(graph_id__in=graphs).annotate(
-            order_field=Case(
-                When(resourceinstanceid__in=initial_values, then=Value(0)),
-                default=Value(1),
-            )
+        resources = ResourceInstance.objects.filter(graph_id__in=graphs).exclude(
+            resourceinstanceid__in=initial_values
         )
 
-        query_string = "descriptors__{}__name__icontains".format(language)
-
         if filter_term:
             resources = resources.filter(
-                Q(**{query_string: filter_term})
-                | Q(resourceinstanceid__in=initial_values)
+                 Q(**{"descriptors__{}__name__icontains".format(language): filter_term})
             )
 
-        paginator = Paginator(
-            resources.order_by(
-                "order_field", "descriptors__{}__name".format(get_language())
-            ),
-            items_per_page,
-        )
-        page_object = paginator.get_page(page_number)
+        offset = (int(page_number) - 1) * int(items_per_page)
+        limit = (int(offset)) + int(items_per_page)
+
         data = [
             {
                 "resourceinstanceid": resource.resourceinstanceid,
                 "display_value": resource.descriptors[language]["name"],
-                "order_field": resource.order_field,
```

Comment:
Removing the sort makes sense when huge sets are in play. Two questions:
- Should we sort by graph so that resource types stay together? I'm showing this sort is still fast b/c indexed.
- Do you want to keep the sort if we can estimate that the resource table is less than humongous?
```sql
SELECT reltuples AS estimate FROM pg_class WHERE relname = 'resource_instances';
   estimate    
---------------
 1.3124129e+07
(1 row)
```

Reply:
Related to this and the other comment, I just ran across:
https://github.com/stephenmcd/django-postgres-fuzzycount

Reply:
Maybe something to riff on rather than install (9 years ago last commit)